File: src/main/java/org/kite9/diagram/visualization/orthogonalization/flow/OrthBuilder.java
package org.kite9.diagram.visualization.orthogonalization.flow;

import org.kite9.diagram.visualization.orthogonalization.OrthogonalizationImpl;
import org.kite9.diagram.visualization.planarization.Planarization;

/**
 * Converts an optimised flow graph and planarization into an orthogonal model.
 * 
 * @author robmoffat
 *
 */
public interface OrthBuilder {

	public abstract OrthogonalizationImpl build(Planarization pln);

}

File: src/main/kotlin/org/kite9/diagram/visualization/orthogonalization/flow/OrthBuilder.kt
package org.kite9.diagram.visualization.orthogonalization.flow

import org.kite9.diagram.visualization.orthogonalization.Orthogonalization
import org.kite9.diagram.visualization.orthogonalization.OrthogonalizationImpl
import org.kite9.diagram.visualization.planarization.Planarization

/**
 * Converts an optimised flow graph and planarization into an orthogonal model.
 *
 * @author robmoffat
 */
interface OrthBuilder {

    fun build(pln: Planarization): Orthogonalization
}